### PR TITLE
fix error message for is.number assertion (closes #2470)

### LINKED ISSUE #2470: Selector API should report NaN instead of integer when type validations failed
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
When `NaN` is passed to the `Selector.nth()` method, type validations throw the following cryptic error:
```
"index" argument is expected to be a number, but it was number.
```

### What is the expected behavior?
Errors with `NaN` error must be reported clearly:
```
"index" argument is expected to be a number, but it was NaN.
```

### How would you reproduce the current behavior (if this is a bug)?
```
const element = Selector('div').nth(NaN)
```

### Specify your

* operating system: any
* testcafe version: 0.20.2
* node.js version: 10.1.0

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/errors/runtime/type-assertions.js [hunk 1]
```diff
@@ -15,8 +15,9 @@ function isNonNegativeValue (value) {
 
 export var is = {
     number: {
-        name:      'number',
-        predicate: isFiniteNumber
+        name:              'number',
+        predicate:         isFiniteNumber,
+        getActualValueMsg: (value, type) => type === 'number' ? value : type
```
Comment (AndreyBelym) [inline src/errors/runtime/type-assertions.js]: Please, make it clear that we filter `NaN` here:
```
getActualValueMsg: (value, type) => isNaN(value) ? value : type
```

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

File: src/errors/runtime/type-assertions.js (hunk 1 above)
Comment (AlexKamaev) [reply src/errors/runtime/type-assertions.js]: `INFINITY` is also possible

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**